motionplan/motionPlanner.go

```diff
@@ -381,3 +382,81 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
```

Comment:
The PR description describes a bunch of cases that this function currently supports. I don't know the motion domain well enough yet to know if there are any usecases it does NOT support.

Would it be a good idea to add the list of use cases it does/does not support in this comment?

Reply:
> if there are any usecases it does NOT support.

in my knowledge, this function has been generalized to the point that it can be used by anything that uses the motion service. Hence, I am unaware of any usecases this function does not support.
@biotinker @raybjork Do you agree?

> Would it be a good idea to add the list of use cases it does/does not support in this comment?

I think it would be smart to list cases this function does not support, but as mentioned I am unaware of any.

Reply:
If it supports everything the motion service supports then I agree a comment is not needed.